library/std/src/net/parser.rs

```diff
@@ -90,7 +92,7 @@ impl<'a> Parser<'a> {
     {
         self.read_atomically(move |p| {
             if index > 0 {
-                let _ = p.read_given_char(sep)?;
+                let () = p.read_given_char(sep)?;
```

Comment:
```suggestion
                p.read_given_char(sep)?;
```

Reply:
These are intentional; without them, a missing `?` at the end of the line would go unnoticed. See https://github.com/rust-lang/rust-clippy/issues/3915

Reply:
Okay but this conflicts with https://rust-lang.github.io/rust-clippy/rust-1.46.0/index.html#let_unit_value. :slightly_frowning_face: I am not a fan. Could we protect against missing `?` by using #\[must_use\] on the relevant functions instead? Someone adding new calls isn't going to remember to write `let ()` if they don't remember to write `?`.

Reply:
You're the boss. Done.